logging/src/main/java/org/jboss/as/logging/LoggingResource.java

```diff
@@ -126,7 +125,19 @@ public Resource requireChild(final PathElement element) {
     @Override
     public boolean hasChildren(final String childType) {
         if (LOG_FILE.equals(childType)) {
-            return !getChildrenNames(LOG_FILE).isEmpty();
+            final String logDir = pathManager.getPathEntry(ServerEnvironment.SERVER_LOG_DIR).resolvePath();
+            if (logDir != null) {
+                final Path dir = Paths.get(logDir);
+                final Collection<String> validFileNames = findValidFileNames(getFileHandlersModel());
+                // If there's at least one child, then we have children
+                for (String name : validFileNames) {
+                    final Path file = dir.resolve(name);
+                    if (Files.exists(file) && Files.isReadable(file)) {
+                        return true;
+                    }
+                }
+            }
+            return false;
```

Comment:
Will this result in no longer reporting children if the base file has been removed but a rotated file is still present? And if so is that ok?